test-integration/server/GraknClientIT.java

```diff
@@ -262,7 +262,7 @@ public void explanationIsGivenWithRule() {
             Set<ConceptMap> deductions = deductions(answer);
             assertTrue(deductions.stream().anyMatch(ans -> {
                 assertNotNull(ans.queryPattern());
-                if (ans.hasExplanation()) {
+                if (ans.hasExplanation() && ans.explanation().getAnswers().size() == 1) {
```

Comment:
what is this special extra check for?

Reply:
This is the non-deterministic failure I was telling you about - the check was previously looking for the single `RuleExplanation` among the answers. Now there are two explanations, a `RuleExplanation` and a `JoinExplanation`, and without this new check it could find either. This test uses the client so we can't check the explanation's type specifically.